src/components/Icon/Direction.tsx

```diff
@@ -0,0 +1,12 @@
+import { SVGProps } from "react";
+
+export enum IconDirection {
+  Up = "up",
+  Down = "down",
+  Left = "left",
+  Right = "right",
+}
+
+export interface IconDirectionProps extends SVGProps<SVGSVGElement> {
+  direction: IconDirection;
+}
```

Comment:
I like that this was moved elsewhere, two things:

- This file could just be a .ts, as we are not using tsx here; it should also be named `IconDirection.ts`;
- Wondering why `IconDirectionProps` is extending `SVGProps`? Can we do in the icons that have direction `const Svg = (props: IconDirectionProps & SVGProps<SVGSVGElement>) => ()`
- Usually when we have string enums, we have them all caps, see here https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

Reply:
resolved with [fix Direction.ts, capitalize Enums, add TODO on banner component](https://github.com/Major-Eth/yearn-web-lib/pull/21/commits/df32714bb98ffa32befd9f149a24aca802303c21)